main/src/main/java/cgeo/geocaching/filters/core/NumberRangeFilter.java

```diff
@@ -35,8 +35,13 @@ public NumberRangeFilter(final Func1<String, T> numberParser, final Func1<Float,
     }
 
     public boolean isInRange(final T value) {
-        if (includeSpecialNumber != null && specialNumber != null && isEqualValue(value, specialNumber)) {
-            return includeSpecialNumber;
+        if (includeSpecialNumber != null && specialNumber != null) {
+            if (isEqualValue(value, specialNumber)) {
+                return includeSpecialNumber;
+            }
+            if (specialNumber.compareTo(value) > 0) { // Special case for lab caches
```

Comment:
This change de facto converts the special number logic from an "equals" logic to a "equal or less than logic".

you need to adjust method "addRangeToSQL" accordingly so generated SQL is also changed from "equals" to "equal or less then" (around code lines 165/165)

Also you're changing a multiple-purpose here for a special case behavior. Was it throughly tested that this new logic also behaves correctly in all other usages of this filter helper?